google-cloud-spanner/src/test/java/com/google/cloud/spanner/SpanTest.java

```diff
@@ -106,6 +106,17 @@ public static void startStaticServer() throws Exception {
         "This test is only supported on JDK11 and lower",
         JavaVersionUtil.getJavaMajorVersion() < 12);
 
+    // Use a little reflection to set the test tracer.
```

Comment:
It caused a test failure and needed to be moved up because:
1. If automatic dialect detection on startup is enabled, the session pool will execute the detection query in the background.
2. That background query could be executed before the `failOnOverkillTraceComponent` was set, which would then cause a NullPointerException, as it would trigger the trace feature.